### PR TITLE
[OSDOCS-12288]: Adding live migration docs

### PR DESCRIPTION
**Pull request by lahinson**
Version(s): 4.17+

Issue: https://issues.redhat.com/browse/OSDOCS-12288

Link to docs preview:
- New prerequisite: https://84924--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-deploy/hcp-deploy-virt.html#hcp-virt-prereqs_hcp-deploy-virt
- New section about live migration: https://84924--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-deploy/hcp-deploy-virt.html#hcp-virt-live-migration_hcp-deploy-virt

QE review:
- [x] QE has approved this change.

Additional information: This PR adds a couple of prerequisites to the docs for hosted control planes on OpenShift Virtualization. It also adds a concept module about live migration. Finally, it reorganizes some content to better meet the mod doc guidelines.

### CODE REVIEW

**Review by davidvossel — COMMENTED** [review]
/lgtm

File: modules/hcp-virt-reqs.adoc [hunk 1]
```diff
@@ -47,6 +47,8 @@ $ oc patch storageclass ocs-storagecluster-ceph-rbd -p '{"metadata": {"annotatio
 $ oc get managedclusters local-cluster
 ----
 
+* On the {product-title} cluster that hosts the {VirtProductName} virtual machines, you must use a `ReadWriteMany` (RWX) storage class so that live migration can be enabled.
```
Comment (zanetworker) [inline modules/hcp-virt-reqs.adoc:50]: Can we add another requisite here that the management cluster can only be on-prem baremetal?

File: modules/hcp-virt-live-migration.adoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+:_mod-docs-content-type: CONCEPT
+[id="hcp-virt-live-migration_{context}"]
+= Live migration for compute nodes
+
+While the management cluster for hosted cluster virtual machines (VMs) is undergoing updates or maintenance, the hosted cluster VMs can be automatically live migrated to prevent disrupting hosted cluster workloads. As a result, the management cluster can be updated without affecting the availability and operation of the KubeVirt platform hosted clusters.
+
+[IMPORTANT]
+====
+The live migration of KubeVirt VMs is enabled by default provided that the VMs use `ReadWriteMany` (RWX) storage for both the root volume and the storage classes that are mapped to `kubevirt-csi`.
```
Comment (maxwelldb) [inline modules/hcp-virt-live-migration.adoc:13]: Might want a noun after `kubevirt-csi`.

File: modules/hcp-virt-prereqs.adoc
Comment (maxwelldb) [inline modules/hcp-virt-prereqs.adoc:1]: I get why in a product sense the prereqs might have to be this heavy. I do wish that there were fewer of them or that we supported checklists in the docs. :weary:

File: modules/hcp-virt-prereqs.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+//
+// * hosted_control_planes/hcp-deploy-virt.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="hcp-virt-prereqs_{context}"]
+= Prerequisites
+
+You must meet the following prerequisites to create an {product-title} cluster on {VirtProductName}:
+
+* You have administrator access to an {product-title} cluster, version 4.14 or later, specified by the `KUBECONFIG` environment variable.
```
Comment (maxwelldb) [inline modules/hcp-virt-prereqs.adoc:11]: The access is specified in the config, or credentials? Nitpicking here.

File: modules/hcp-virt-prereqs.adoc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+* The {product-title} management cluster has a default storage class. For more information, see "Postinstallation storage configuration". The following example shows how to set a default storage class:
++
+[source,terminal]
+----
+$ oc patch storageclass ocs-storagecluster-ceph-rbd -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'
+----
+
+* You have a valid pull secret file for the `quay.io/openshift-release-dev` repository. For more information, see "Install OpenShift on any x86_64 platform with user-provisioned infrastructure".
+* You have installed the hosted control plane command line interface.
+* You configured a load balancer. For more information, see "Optional: Configuring MetalLB".
```
Comment (maxwelldb) [inline modules/hcp-virt-prereqs.adoc:30]: Is it important to mix present perfect and preterite? Standardize if not.

**Review by maxwelldb — APPROVED** [review]
Looks pretty good to me!

File: modules/hcp-virt-prereqs.adoc
Comment (lahinson) [reply modules/hcp-virt-prereqs.adoc:1]: Agree 1000%

File: modules/hcp-virt-live-migration.adoc (hunk 2 above)
Comment (lahinson) [reply modules/hcp-virt-live-migration.adoc:13]: Will do

**Pull request merged**